### ISSUE TITLE
Unable to maximize windows on Windows 10

### ISSUE THREAD

**Issue by marovira** [glfw/glfw#1612]
OS: Windows 10 64 bit.
Release: GLFW 3.3
Compiled with Visual Studio 2019 16.4.1
Error logs: The program reports no errors.

Setting `glfwWindowHint(GLFW_MAXIMIZED, GLFW_TRUE)` or calling `glfwMaximizeWindow(window)` does not cause the window to be maximized. Instead, the window retains the size it was given upon creation. Below is a minimum working example that shows the problem:

```c++
#include <GL/gl3w.h>
#include <GLFW/glfw3.h>

#include <iostream>

void onError(int code, char const* message)
{
   std::cout << "error (" << code << "): " << message << std::endl;
}

int main()
{
   glfwSetErrorCallback(onError);
   glfwInit();

   glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 4);
   glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 5);
   glfwWindowHint(GLFW_OPENGL_PROFILE, GLFW_OPENGL_CORE_PROFILE);
   glfwWindowHint(GLFW_OPENGL_FORWARD_COMPAT, GLFW_TRUE);
   glfwWindowHint(GLFW_RESIZABLE, GLFW_TRUE);

   // This does not work, the window is shown not maximized.
   glfwWindowHint(GLFW_MAXIMIZED, GLFW_TRUE);

   GLFWwindow* window = glfwCreateWindow(500, 500, "Demo", nullptr, nullptr);
   glfwMakeContextCurrent(window);
   glfwSwapInterval(1);

   gl3wInit();

   // This does not work either.
   glfwMaximizeWindow(window);

   while (!glfwWindowShouldClose(window))
   {
      int width, height;
      glfwGetFramebufferSize(window, &width, &height);
      glViewport(0, 0, width, height);
      glClearColor(0, 0, 0, 0);
      glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);

      glfwSwapBuffers(window);
      glfwPollEvents();
   }

   glfwDestroyWindow(window);
   glfwTerminate();
}
```
If you require any further information please let me know.

**Comment by elmindreda** [comment]
Thank you for an excellent bug report! This should be the same issue that was fixed with #1499. The fix is in `master`. In the meantime on version 3.3 maximization with `glfwMaximizeWindow` should work again if you remove the window hint.

**Comment by marovira** [comment]
Thanks for the quick reply! I can confirm that this solves the issue and I'll update my GLFW versions accordingly. Thanks!
